inference-engine/src/mkldnn_plugin/nodes/bucketize.cpp

```diff
@@ -72,46 +77,99 @@ class BucketizeImpl : public ExtLayerBase {
     }
 
     StatusCode execute(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs, ResponseDesc *resp) noexcept override {
-        const float *input_tensor_ptr = inputs[INPUT_TENSOR_PORT]->cbuffer().as<const float *>() +
-            inputs[INPUT_TENSOR_PORT]->getTensorDesc().getBlockingDesc().getOffsetPadding();
-        const float *input_bins_ptr = nullptr;
-        if (with_bins) {
-            input_bins_ptr = inputs[INPUT_BINS_PORT]->cbuffer().as<const float *>() +
-                inputs[INPUT_BINS_PORT]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+        if (input_precision == Precision::FP32 && boundaries_precision == Precision::FP32 &&
```

Comment:
nit: There is this method for combining several previsions that could be passed to switch expression. It would make code readable. 
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/include/ie_precision.hpp#L411

Usage example: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/legacy_api/src/shape_infer/const_infer/ie_convert_const_infer.hpp#L94

Reply:
Nice proposal. Redone.